### PR TITLE
fix: fix entity specs

### PR DESCRIPTION
**Pull request by UziTech**
**Marked version:** 4.2.2

## Description

Fix `Entity and numeric character references` specs [#28](https://spec.commonmark.org/0.30/#example-28) and [#34](https://spec.commonmark.org/0.30/#example-34).

In order to fix the last 2 `Entity and numeric character references` specs we would need to convert html entities to percent encoded with this json map: https://html.spec.whatwg.org/entities.json. This would increase the size of marked significantly (almost double 48k -> 84k) and provide very little benefit to the actual output.

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

**Review by calculuschild — APPROVED** [review]

**Comment by calculuschild** [comment]
Is there really no built-in JS function to convert html entities?

**Comment by UziTech** [comment]
Unfortunately no. The reason given is because they are HTML entities not JavaScript entities. 🫤

**Comment by UziTech** [comment]
Any objections to merging this?

**Review by calculuschild — APPROVED** [review]

**Comment by styfle** [comment]
> This would increase the size of marked significantly (almost double 48k -> 84k)

That would be unfortunate. Perhaps we could add the most common ones to `escapeReplacements`?

File: src/Renderer.js [hunk 1]
```diff
@@ -170,7 +170,7 @@ export class Renderer {
     if (href === null) {
       return text;
     }
-    let out = '<a href="' + escape(href) + '"';
+    let out = '<a href="' + href + '"';
```
Comment (styfle) [inline src/Renderer.js:173]: What about when `href` contains a `'` or `"`? 

Do we have a test for that?

Comment (UziTech) [reply src/Renderer.js:173]: the href runs through `encodeURI` in the `cleanURL`

https://github.com/UziTech/marked/blob/57a203182cddb4dc1ae4ac8f7c72650eddf3442d/src/helpers.js#L98

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**